public/app/features/explore/LogsSamplePanel.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+
+import { CoreApp, DataQueryResponse, DataSourceApi, LoadingState, LogsDedupStrategy, TimeZone } from '@grafana/data';
+import { reportInteraction } from '@grafana/runtime';
+import { Collapse } from '@grafana/ui';
+import { dataFrameToLogsModel } from 'app/core/logsModel';
+import store from 'app/core/store';
+
+import { LogRows } from '../logs/components/LogRows';
+
+import { SupplementaryResultError } from './SupplementaryResultError';
+import { SETTINGS_KEYS } from './utils/logs';
+
+type Props = {
+  data: DataQueryResponse | undefined;
+  enabled: boolean;
+  timeZone: TimeZone;
+  datasourceInstance: DataSourceApi | null | undefined;
+  setLogsSampleEnabled: (enabled: boolean) => void;
+};
+
+export function LogsSamplePanel(props: Props) {
+  const { data, timeZone, enabled, setLogsSampleEnabled, datasourceInstance } = props;
+
+  const onToggleLogsSampleCollapse = (isOpen: boolean) => {
+    setLogsSampleEnabled(isOpen);
+    reportInteraction('grafana_explore_logs_sample_toggle_clicked', {
+      datasourceType: datasourceInstance ? datasourceInstance?.type : 'unknown',
+      type: isOpen ? 'open' : 'close',
+    });
+  };
+
+  let LogsSamplePanelContent: JSX.Element | null;
+
+  if (data === undefined) {
+    LogsSamplePanelContent = null;
+  } else if (data?.error !== undefined) {
+    LogsSamplePanelContent = (
+      <SupplementaryResultError error={data.error} title="Failed to load log samples for this query" />
+    );
+  } else if (data?.state === LoadingState.Loading) {
+    LogsSamplePanelContent = <span>Log samples are loading...</span>;
+  } else if (data?.data.length === 0 || data?.data[0].length === 0) {
+    LogsSamplePanelContent = <span>No logs sample data.</span>;
+  } else {
+    const logs = dataFrameToLogsModel(data.data, undefined);
```

Comment:
I totally agree. I removed intervalMs required parameter and changed it to optional (as it could have `undefined` 
value) in https://github.com/grafana/grafana/pull/61864/commits/a2b03a8fff58dd9fc47e0ef3523afb58969691bc. I also updated documentation for this function to give it more context on what is used and why. And removed all places where we had to use `undefined` argument when calling this function.